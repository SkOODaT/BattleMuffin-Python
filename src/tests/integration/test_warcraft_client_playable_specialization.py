import json
import os

from clients.warcraft_client import WarcraftClient


def test_get_playable_specializations_index(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_playable_specializations_index()
    assert response == snapshot


def test_get_playable_specialization(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_playable_specialization(262)
    text = json.dumps(response)
    text = text.replace("\\r\\n", "\\n")
    obj = json.loads(text)
    assert obj == snapshot


def test_get_playable_specialization_media(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_playable_specialization_media(262)
    assert response == snapshot
