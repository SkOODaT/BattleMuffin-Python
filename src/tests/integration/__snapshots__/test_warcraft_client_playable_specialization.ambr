# name: test_get_playable_specialization
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/262?namespace=static-8.3.0_32861-us',
      },
    },
    'gender_description': <class 'dict'> {
      'female': '
  A spellcaster who harnesses the destructive forces of nature and the elements.
  
  Preferred Weapon: Mace, Dagger, and Shield    
  ',
      'male': '
  A spellcaster who harnesses the destructive forces of nature and the elements.
  
  Preferred Weapon: Mace, Dagger, and Shield    
  ',
    },
    'id': 262,
    'media': <class 'dict'> {
      'id': 262,
      'key': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/playable-specialization/262?namespace=static-8.3.0_32861-us',
      },
    },
    'name': 'Elemental',
    'playable_class': <class 'dict'> {
      'id': 7,
      'key': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/playable-class/7?namespace=static-8.3.0_32861-us',
      },
      'name': 'Shaman',
    },
    'pvp_talents': <class 'list'> [
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Passive',
          'description': 'Increases your maximum Maelstrom by 50.',
        },
        'talent': <class 'dict'> {
          'id': 727,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/727?namespace=static-8.3.0_32861-us',
          },
          'name': 'Elemental Attunement',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Passive',
          'description': '
  Lava Burst damage increased by 25%.
  
  Your Flame Shock damage over time has a 10% chance to reset the remaining cooldown on Lava Burst, and increase its damage by 25%, stacking up to 3 times.        
  ',
        },
        'talent': <class 'dict'> {
          'id': 728,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/728?namespace=static-8.3.0_32861-us',
          },
          'name': 'Control of Lava',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Passive',
          'description': 'Thunderstorm now can be cast on allies within 40 yards, and its cooldown is reduced by 15 sec.',
        },
        'talent': <class 'dict'> {
          'id': 730,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/730?namespace=static-8.3.0_32861-us',
          },
          'name': 'Traveling Storms',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Instant',
          'cooldown': '30 sec cooldown',
          'description': 'Grips the target in lightning, stunning and dealing 38,960 Nature damage over 5 sec while the target is lassoed. Can move while channeling.',
          'range': '20 yd range',
        },
        'talent': <class 'dict'> {
          'id': 731,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/731?namespace=static-8.3.0_32861-us',
          },
          'name': 'Lightning Lasso',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Passive',
          'description': '
  While in Ghost Wolf, you heal 3% health every 2 sec.
  
  Increases the movement speed of Ghost Wolf by an additional 10%.        
  ',
        },
        'talent': <class 'dict'> {
          'id': 3062,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/3062?namespace=static-8.3.0_32861-us',
          },
          'name': 'Spectral Recovery',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Instant',
          'cooldown': '40 sec cooldown',
          'description': 'Summons a Fire Totem with 5 health at the feet of the caster for 15 sec that increases the critical effect of damage and healing spells of all nearby allies within 40 yards by 20% for 15 sec.',
          'power_cost': '600 Mana',
        },
        'talent': <class 'dict'> {
          'id': 3488,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/3488?namespace=static-8.3.0_32861-us',
          },
          'name': 'Skyfury Totem',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Instant',
          'cooldown': '45 sec cooldown',
          'description': '
  Summons an Air Totem with 5 health at the feet of the caster for 15 sec. 
  
  Whenever enemies within 20 yards of the totem deal direct damage, the totem will deal 100% of the damage dealt back to attacker.         
  ',
          'power_cost': '600 Mana',
          'range': '40 yd range',
        },
        'talent': <class 'dict'> {
          'id': 3490,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/3490?namespace=static-8.3.0_32861-us',
          },
          'name': 'Counterstrike Totem',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Passive',
          'description': 'Whenever you successfully Purge a target or when you are dispelled, you are healed for 3% of your total health.',
        },
        'talent': <class 'dict'> {
          'id': 3491,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/3491?namespace=static-8.3.0_32861-us',
          },
          'name': 'Purifying Waters',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Passive',
          'description': 'Duration of incoming crowd control effects reduced by 20%.  Does not stack with similar effects.',
        },
        'talent': <class 'dict'> {
          'id': 3596,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/3596?namespace=static-8.3.0_32861-us',
          },
          'name': 'Relentless',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Passive',
          'description': 'Removes any loss of control effect with a duration of 5 seconds or more. This effect can only occur once every 1 min.',
        },
        'talent': <class 'dict'> {
          'id': 3597,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/3597?namespace=static-8.3.0_32861-us',
          },
          'name': 'Adaptation',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Instant',
          'cooldown': '2 min cooldown',
          'description': 'Removes all movement impairing effects and all effects which cause loss of control of your character while in PvP combat.',
        },
        'talent': <class 'dict'> {
          'id': 3598,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/3598?namespace=static-8.3.0_32861-us',
          },
          'name': "Gladiator's Medallion",
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Instant',
          'cooldown': '30 sec cooldown',
          'description': 'Summons an Air Totem with 5 health at the feet of the caster that will redirect all harmful spells cast on a nearby party or raid member to itself. Will not redirect area of effect spells. Lasts 3 sec.',
          'power_cost': '1,180 Mana',
        },
        'talent': <class 'dict'> {
          'id': 3620,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/3620?namespace=static-8.3.0_32861-us',
          },
          'name': 'Grounding Totem',
        },
      },
      <class 'dict'> {
        'spell_tooltip': <class 'dict'> {
          'cast_time': 'Passive',
          'description': 'When you cast Healing Surge on yourself, you are healed for 50% of the amount 3 sec later.',
        },
        'talent': <class 'dict'> {
          'id': 3621,
          'key': <class 'dict'> {
            'href': 'https://us.api.blizzard.com/data/wow/pvp-talent/3621?namespace=static-8.3.0_32861-us',
          },
          'name': 'Swelling Waves',
        },
      },
    ],
    'role': <class 'dict'> {
      'name': 'Damage',
      'type': 'DAMAGE',
    },
    'talent_tiers': <class 'list'> [
      <class 'dict'> {
        'level': 15,
        'talents': <class 'list'> [
          <class 'dict'> {
            'column_index': 0,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': 'Your damaging spells incite the earth around you to come to your aid for 6 sec, repeatedly dealing 833 Nature damage to your most recently attacked target.',
            },
            'talent': <class 'dict'> {
              'id': 22356,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/22356?namespace=static-8.3.0_32861-us',
              },
              'name': 'Earthen Rage',
            },
          },
          <class 'dict'> {
            'column_index': 1,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': 'Lava Burst now has 2 charges. Effects that reset its remaining cooldown will instead grant 1 charge.',
            },
            'talent': <class 'dict'> {
              'id': 22357,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/22357?namespace=static-8.3.0_32861-us',
              },
              'name': 'Echo of the Elements',
            },
          },
          <class 'dict'> {
            'column_index': 2,
            'spell_tooltip': <class 'dict'> {
              'cast_time': '2 sec cast',
              'cooldown': '12 sec cooldown',
              'description': '
  Harnesses the raw power of the elements, dealing 3,818 Elemental damage and increasing your Critical Strike, Haste, or Mastery by 366 for 10 sec.
  
  Can cause an Elemental Overload.            
  ',
              'range': '40 yd range',
            },
            'talent': <class 'dict'> {
              'id': 22358,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/22358?namespace=static-8.3.0_32861-us',
              },
              'name': 'Elemental Blast',
            },
          },
        ],
        'tier_index': 0,
      },
      <class 'dict'> {
        'level': 30,
        'talents': <class 'list'> [
          <class 'dict'> {
            'column_index': 0,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': 'Earth Shock and Earthquake have a 25% chance to refund the Maelstrom spent on them.',
            },
            'talent': <class 'dict'> {
              'id': 23108,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/23108?namespace=static-8.3.0_32861-us',
              },
              'name': 'Aftershock',
            },
          },
          <class 'dict'> {
            'column_index': 1,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': 'Your maximum Maelstrom increased by 30, and the Maelstrom cost of Earth Shock and Earthquake is reduced by 10.',
            },
            'talent': <class 'dict'> {
              'id': 22139,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/22139?namespace=static-8.3.0_32861-us',
              },
              'name': 'Call the Thunder',
            },
          },
          <class 'dict'> {
            'column_index': 2,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Instant',
              'description': '
  Summons four totems that increase your combat capabilities for 2 min.
  
  Resonance Totem
  Generates 1 Maelstrom every 1 sec.
  
  Storm Totem
  Increases the chance for Lightning Bolt and Chain Lightning to trigger Elemental Overload by 5%.
  
  Ember Totem
  Increases Flame Shock damage over time by 10%.
  
  Tailwind Totem
  Increases your Haste by 2%.            
  ',
              'range': '40 yd range',
            },
            'talent': <class 'dict'> {
              'id': 23190,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/23190?namespace=static-8.3.0_32861-us',
              },
              'name': 'Totem Mastery',
            },
          },
        ],
        'tier_index': 1,
      },
      <class 'dict'> {
        'level': 45,
        'talents': <class 'list'> [
          <class 'dict'> {
            'column_index': 0,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': 'While transformed into a Ghost Wolf, you gain 5% increased movement speed and 5% damage reduction every 1 sec, stacking up to 4 times.',
            },
            'talent': <class 'dict'> {
              'id': 23162,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/23162?namespace=static-8.3.0_32861-us',
              },
              'name': 'Spirit Wolf',
            },
          },
          <class 'dict'> {
            'column_index': 1,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Instant',
              'description': '
  Protects the target with an earthen shield, increasing your healing on them by 10% and healing them for 3,401 when they take damage. 9 charges. This heal can only occur once every few seconds. 
  
  Earth Shield can only be placed on one target at a time.            
  ',
              'power_cost': '2,000 Mana',
              'range': '40 yd range',
            },
            'talent': <class 'dict'> {
              'id': 23163,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/23163?namespace=static-8.3.0_32861-us',
              },
              'name': 'Earth Shield',
            },
          },
          <class 'dict'> {
            'column_index': 2,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'cooldown': '1 min cooldown',
              'description': 'Reduces the cooldown of Capacitor Totem by 5 sec for each enemy it stuns, up to a maximum reduction of 20 sec.',
            },
            'talent': <class 'dict'> {
              'id': 23164,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/23164?namespace=static-8.3.0_32861-us',
              },
              'name': 'Static Charge',
            },
          },
        ],
        'tier_index': 2,
      },
      <class 'dict'> {
        'level': 60,
        'talents': <class 'list'> [
          <class 'dict'> {
            'column_index': 0,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': 'Casting Lava Burst increases the damage of your next Nature, Physical, or Frost spell by 20%.',
            },
            'talent': <class 'dict'> {
              'id': 19271,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/19271?namespace=static-8.3.0_32861-us',
              },
              'name': 'Master of the Elements',
            },
          },
          <class 'dict'> {
            'column_index': 1,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Instant',
              'description': '
  Calls forth a Greater Storm Elemental to hurl gusts of wind that damage the Shaman's enemies for 30 sec.
  
  While the Storm Elemental is active, each time you cast Lightning Bolt or Chain Lightning, the cast time of Lightning Bolt and Chain Lightning is reduced by 3%, stacking up to 20 times.            
  ',
              'range': '40 yd range',
            },
            'talent': <class 'dict'> {
              'id': 19272,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/19272?namespace=static-8.3.0_32861-us',
              },
              'name': 'Storm Elemental',
            },
          },
          <class 'dict'> {
            'column_index': 2,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Instant',
              'cooldown': '1 min cooldown',
              'description': 'Summons a totem at the target location for 15 sec that hurls liquid magma at a random nearby target every 0.9 sec, dealing 909 Fire damage to all enemies within 8 yards.',
              'range': '40 yd range',
            },
            'talent': <class 'dict'> {
              'id': 19273,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/19273?namespace=static-8.3.0_32861-us',
              },
              'name': 'Liquid Magma Totem',
            },
          },
        ],
        'tier_index': 3,
      },
      <class 'dict'> {
        'level': 75,
        'talents': <class 'list'> [
          <class 'dict'> {
            'column_index': 0,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': 'When your health is brought below 35%, you instantly heal for 20% of your maximum health.  Cannot occur more than once every 45 sec.',
            },
            'talent': <class 'dict'> {
              'id': 22144,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/22144?namespace=static-8.3.0_32861-us',
              },
              'name': "Nature's Guardian",
            },
          },
          <class 'dict'> {
            'column_index': 1,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Instant',
              'cooldown': '2 min cooldown',
              'description': 'For the next 10 sec, 25% of your damage and healing is converted to healing on up to 3 nearby injured party or raid members.',
            },
            'talent': <class 'dict'> {
              'id': 22172,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/22172?namespace=static-8.3.0_32861-us',
              },
              'name': 'Ancestral Guidance',
            },
          },
          <class 'dict'> {
            'column_index': 2,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Instant',
              'cooldown': '2 min cooldown',
              'description': 'Summons a totem at the target location for 15 sec, continually granting all allies who pass within 10 yards 60% increased movement speed for 5 sec.',
              'range': '40 yd range',
            },
            'talent': <class 'dict'> {
              'id': 21966,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/21966?namespace=static-8.3.0_32861-us',
              },
              'name': 'Wind Rush Totem',
            },
          },
        ],
        'tier_index': 4,
      },
      <class 'dict'> {
        'level': 90,
        'talents': <class 'list'> [
          <class 'dict'> {
            'column_index': 0,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': '
  Earth Shock also enhances your next spell cast within 15 sec:
  
  Flame Shock: The next cast also applies Flame Shock to 1 additional target within 8 yards of the target.
  Lightning Bolt: Your next cast will cause an additional 1-3 Elemental Overloads.
  Lava Burst: Reduces the cooldown of your Fire Elemental by 6.0 sec.
  Frost Shock: Freezes the target in place for 6 sec.
              
  ',
            },
            'talent': <class 'dict'> {
              'id': 22145,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/22145?namespace=static-8.3.0_32861-us',
              },
              'name': 'Surge of Power',
            },
          },
          <class 'dict'> {
            'column_index': 1,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': 'Your Earth, Fire, and Storm Elementals are drawn from primal elementals 80% more powerful than regular elementals, with additional abilities, and you gain direct control over them.',
            },
            'talent': <class 'dict'> {
              'id': 19266,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/19266?namespace=static-8.3.0_32861-us',
              },
              'name': 'Primal Elementalist',
            },
          },
          <class 'dict'> {
            'column_index': 2,
            'spell_tooltip': <class 'dict'> {
              'cast_time': '2 sec cast',
              'cooldown': '30 sec cooldown',
              'description': '
  Hurls frigid ice at the target, dealing 3,333 Frost damage and causing your next 4 Frost Shocks to deal 200% increased damage and generate 8 Maelstrom.
  
  Generates 25 Maelstrom.
  
  Can cause an Elemental Overload.            
  ',
              'range': '40 yd range',
            },
            'talent': <class 'dict'> {
              'id': 23111,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/23111?namespace=static-8.3.0_32861-us',
              },
              'name': 'Icefury',
            },
          },
        ],
        'tier_index': 5,
      },
      <class 'dict'> {
        'level': 100,
        'talents': <class 'list'> [
          <class 'dict'> {
            'column_index': 0,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Passive',
              'description': 'When your spells cause an elemental overload, you gain 2% Haste for 10 sec. Gaining a stack does not refresh the duration.',
            },
            'talent': <class 'dict'> {
              'id': 21198,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/21198?namespace=static-8.3.0_32861-us',
              },
              'name': 'Unlimited Power',
            },
          },
          <class 'dict'> {
            'column_index': 1,
            'spell_tooltip': <class 'dict'> {
              'cast_time': '1.5 sec cast',
              'cooldown': '1 min cooldown',
              'description': 'Charge yourself with lightning, causing your next 2 Lightning Bolts to deal 150% more damage, and also causes your next 2 Lightning Bolts or Chain Lightnings to be instant cast and trigger an Elemental Overload on every target.',
            },
            'talent': <class 'dict'> {
              'id': 22153,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/22153?namespace=static-8.3.0_32861-us',
              },
              'name': 'Stormkeeper',
            },
          },
          <class 'dict'> {
            'column_index': 2,
            'spell_tooltip': <class 'dict'> {
              'cast_time': 'Instant',
              'cooldown': '3 min cooldown',
              'description': 'Transform into a Flame Ascendant for 15 sec, replacing Chain Lightning with Lava Beam, removing the cooldown on Lava Burst, and increasing the damage of Lava Burst by an amount equal to your critical strike chance.',
            },
            'talent': <class 'dict'> {
              'id': 21675,
              'key': <class 'dict'> {
                'href': 'https://us.api.blizzard.com/data/wow/talent/21675?namespace=static-8.3.0_32861-us',
              },
              'name': 'Ascendance',
            },
          },
        ],
        'tier_index': 6,
      },
    ],
  }
---
# name: test_get_playable_specialization_media
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/playable-specialization/262?namespace=static-8.3.0_32861-us',
      },
    },
    'assets': <class 'list'> [
      <class 'dict'> {
        'key': 'icon',
        'value': 'https://render-us.worldofwarcraft.com/icons/56/spell_nature_lightning.jpg',
      },
    ],
    'id': 262,
  }
---
# name: test_get_playable_specializations_index
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/?namespace=static-8.3.0_32861-us',
      },
    },
    'character_specializations': <class 'list'> [
      <class 'dict'> {
        'id': 62,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/62?namespace=static-8.3.0_32861-us',
        },
        'name': 'Arcane',
      },
      <class 'dict'> {
        'id': 63,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/63?namespace=static-8.3.0_32861-us',
        },
        'name': 'Fire',
      },
      <class 'dict'> {
        'id': 64,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/64?namespace=static-8.3.0_32861-us',
        },
        'name': 'Frost',
      },
      <class 'dict'> {
        'id': 65,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/65?namespace=static-8.3.0_32861-us',
        },
        'name': 'Holy',
      },
      <class 'dict'> {
        'id': 66,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/66?namespace=static-8.3.0_32861-us',
        },
        'name': 'Protection',
      },
      <class 'dict'> {
        'id': 70,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/70?namespace=static-8.3.0_32861-us',
        },
        'name': 'Retribution',
      },
      <class 'dict'> {
        'id': 71,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/71?namespace=static-8.3.0_32861-us',
        },
        'name': 'Arms',
      },
      <class 'dict'> {
        'id': 72,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/72?namespace=static-8.3.0_32861-us',
        },
        'name': 'Fury',
      },
      <class 'dict'> {
        'id': 73,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/73?namespace=static-8.3.0_32861-us',
        },
        'name': 'Protection',
      },
      <class 'dict'> {
        'id': 102,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/102?namespace=static-8.3.0_32861-us',
        },
        'name': 'Balance',
      },
      <class 'dict'> {
        'id': 103,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/103?namespace=static-8.3.0_32861-us',
        },
        'name': 'Feral',
      },
      <class 'dict'> {
        'id': 104,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/104?namespace=static-8.3.0_32861-us',
        },
        'name': 'Guardian',
      },
      <class 'dict'> {
        'id': 105,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/105?namespace=static-8.3.0_32861-us',
        },
        'name': 'Restoration',
      },
      <class 'dict'> {
        'id': 250,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/250?namespace=static-8.3.0_32861-us',
        },
        'name': 'Blood',
      },
      <class 'dict'> {
        'id': 251,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/251?namespace=static-8.3.0_32861-us',
        },
        'name': 'Frost',
      },
      <class 'dict'> {
        'id': 252,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/252?namespace=static-8.3.0_32861-us',
        },
        'name': 'Unholy',
      },
      <class 'dict'> {
        'id': 253,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/253?namespace=static-8.3.0_32861-us',
        },
        'name': 'Beast Mastery',
      },
      <class 'dict'> {
        'id': 254,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/254?namespace=static-8.3.0_32861-us',
        },
        'name': 'Marksmanship',
      },
      <class 'dict'> {
        'id': 255,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/255?namespace=static-8.3.0_32861-us',
        },
        'name': 'Survival',
      },
      <class 'dict'> {
        'id': 256,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/256?namespace=static-8.3.0_32861-us',
        },
        'name': 'Discipline',
      },
      <class 'dict'> {
        'id': 257,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/257?namespace=static-8.3.0_32861-us',
        },
        'name': 'Holy',
      },
      <class 'dict'> {
        'id': 258,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/258?namespace=static-8.3.0_32861-us',
        },
        'name': 'Shadow',
      },
      <class 'dict'> {
        'id': 259,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/259?namespace=static-8.3.0_32861-us',
        },
        'name': 'Assassination',
      },
      <class 'dict'> {
        'id': 260,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/260?namespace=static-8.3.0_32861-us',
        },
        'name': 'Outlaw',
      },
      <class 'dict'> {
        'id': 261,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/261?namespace=static-8.3.0_32861-us',
        },
        'name': 'Subtlety',
      },
      <class 'dict'> {
        'id': 262,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/262?namespace=static-8.3.0_32861-us',
        },
        'name': 'Elemental',
      },
      <class 'dict'> {
        'id': 263,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/263?namespace=static-8.3.0_32861-us',
        },
        'name': 'Enhancement',
      },
      <class 'dict'> {
        'id': 264,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/264?namespace=static-8.3.0_32861-us',
        },
        'name': 'Restoration',
      },
      <class 'dict'> {
        'id': 265,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/265?namespace=static-8.3.0_32861-us',
        },
        'name': 'Affliction',
      },
      <class 'dict'> {
        'id': 266,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/266?namespace=static-8.3.0_32861-us',
        },
        'name': 'Demonology',
      },
      <class 'dict'> {
        'id': 267,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/267?namespace=static-8.3.0_32861-us',
        },
        'name': 'Destruction',
      },
      <class 'dict'> {
        'id': 268,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/268?namespace=static-8.3.0_32861-us',
        },
        'name': 'Brewmaster',
      },
      <class 'dict'> {
        'id': 269,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/269?namespace=static-8.3.0_32861-us',
        },
        'name': 'Windwalker',
      },
      <class 'dict'> {
        'id': 270,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/270?namespace=static-8.3.0_32861-us',
        },
        'name': 'Mistweaver',
      },
      <class 'dict'> {
        'id': 577,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/577?namespace=static-8.3.0_32861-us',
        },
        'name': 'Havoc',
      },
      <class 'dict'> {
        'id': 581,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/581?namespace=static-8.3.0_32861-us',
        },
        'name': 'Vengeance',
      },
    ],
    'pet_specializations': <class 'list'> [
      <class 'dict'> {
        'id': 74,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/74?namespace=static-8.3.0_32861-us',
        },
        'name': 'Ferocity',
      },
      <class 'dict'> {
        'id': 79,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/79?namespace=static-8.3.0_32861-us',
        },
        'name': 'Cunning',
      },
      <class 'dict'> {
        'id': 81,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/playable-specialization/81?namespace=static-8.3.0_32861-us',
        },
        'name': 'Tenacity',
      },
    ],
  }
---
